src/vs/workbench/contrib/files/browser/files.contribution.ts

```diff
@@ -362,6 +362,15 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('fileDialogDefaultPath', "Default path for file dialogs, overriding user's home path. Only used in the absence of a context-specific path, such as most recently opened file or folder."),
 			'scope': ConfigurationScope.MACHINE
 		},
+		'files.dialog.openFolderStartLocation': {
```

Comment:
The setting name confuses me a bit. Is it used only when opening a folder or also when picking a location to save? what is a "start location"?

Reply:
It is just for when doing "open folder". Is `files.dialog.openFolderStartPath` clearer?

Reply:
But the setting is not allowing you to set the location, it is allowing to configure if the `path` should be used or `dirname(path)`.

Reply:
I have changed the setting back to a boolean and named it `files.dialog.useParentPathForOpenFolder`. Do you prefer that? If not, do you have some suggestions for what it could be instead?